### PR TITLE
feat(protocol-designer): show file upload errors in modal

### LINKED ISSUE #1610: File Management: Upload Errors
As a user if I can't open my file in PD I'd like to know why

## Acceptance criteria
- A modal in response to uploading a file in any of the following scenarios:

## Scenarios + Designs
- [ ]  **Incorrect file type**
_Title_: Incorrect file type
_Body_: 
- Only JSON files created in the Protocol Designer can be imported. 
- At this time Python protocol files are not supported.

- [ ]  **Invalid JSON file**
_Title_: This JSON file cannot be opened
_Body_: 
This file is either missing information it needs to import properly or contains sections that the Protocol Designer cannot read. 
- At this time we do not support JSON files created outside of the Protocol Designer
- If this is a Protocol Designer file that you have edited outside of this tool you may have introduced errors.

### PR DESCRIPTION
**Pull request by IanLondon**
## overview

Closes #1610

## changelog

- `load-file` atom has its first and currently only reducer, which holds file upload errors and is used by `FileUploadErrorModal`
- `loadFile` action converted from a simple action to a thunk, which handles file upload & parsing. It takes the `onChange` event from an `<input type='file' />` component, and dispatches either FILE_ERRORS or LOAD_FILE action.
- show file upload errors in new modal

## review requests

- Uploading a file that doesn't end in `*.json` will give you the "incorrect file type" error
- Uploading a malformed JSON protocol (or anything that ends in `*.json` that throws an error during parsing and LOAD_FILE action) will give you the "Invalid JSON file" error, and show the error message that was thrown
- Uploading a valid JSON protocol (eg one you create and export in PD via this branch) will set your metadata fields & pipettes (and not other stuff, until #1804 is merged)

### CODE REVIEW

File: protocol-designer/src/components/FileSidebar/index.js [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        } catch (error) {
+          this.setState({
+            uploadErrorModal: {
+              errorType: 'INVALID_JSON_FILE',
+              errorMessage: error.message
+            }
+          })
+        }
+      }
+      reader.readAsText(file)
```
Comment (mcous) [inline protocol-designer/src/components/FileSidebar/index.js]: I think this component mixes too many concerns (display + reading and parsing JSON). Can we either:

1. Split into a <SidePanel> component and a DOM-less JSON file reader component
2. (My preference) move file reading to middleware

File: protocol-designer/src/containers/ConnectedFileSidebar.js [hunk 2]
```diff
@@ -1,73 +0,0 @@
-// @flow
```
Comment (IanLondon) [inline protocol-designer/src/containers/ConnectedFileSidebar.js]: Moved to protocol-designer/src/components/FileSidebar/ConnectedFileSidebar.js -- and removed `onUpload` stuff from here

File: protocol-designer/src/components/FileSidebar/FileSidebar.js [hunk 3]
```diff
@@ -1,11 +1,15 @@
...
+  PrimaryButton,
+  OutlineButton,
+  SidePanel
+} from '@opentrons/components'
 import styles from './FileSidebar.css'
 
 type Props = {
-  onUpload: (SyntheticInputEvent<HTMLInputElement>) => mixed,
+  loadFile: (event: SyntheticInputEvent<HTMLInputElement>) => mixed,
   onCreateNew?: () => mixed,
```
Comment (b-cooper) [inline protocol-designer/src/components/FileSidebar/FileSidebar.js]: Non-blocking, just continuing the dicsussion that @mcous brought up earlier. I would vote to have the action dispatch function named `createNewFile` that way `onClick={this.props.createNewFile}` reads semantically as "on clicking this component, create a new file"

File: protocol-designer/src/components/modals/FileUploadErrorModal/FileUploadErrorModal.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+// @flow
+import * as React from 'react'
+import {AlertModal} from '@opentrons/components'
+import modalStyles from '../modal.css'
+import getModalContents from './modalContents'
+import type {FileError} from '../../../load-file'
+
+type Props = {
+  error: ?FileError,
+  onClose: (SyntheticEvent<*>) => mixed
```
Comment (b-cooper) [inline protocol-designer/src/components/modals/FileUploadErrorModal/FileUploadErrorModal.js]: Same comment as above (`close()` over `onClose`)

**Pull request merged**